module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,70 @@
+#include "compression.hpp"
+
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+CompressedPGMBitMap compressGrayscale(const PGMBitMap& img) {
+    CompressedPGMBitMap bitMap;
+    bitMap.reserve(height);
+
+    for (const auto& row : img) {
+        Pixel lastValue = row[0];
+        WidthType consecutiveCount = 1;
+
+        for (uint8_t i = 1; i < width; i++) {
+            if (row[i] != lastValue) {
+                bitMap.emplace_back(std::make_pair(lastValue, consecutiveCount));
+                lastValue = row[i];
+                consecutiveCount = 1;
+            } else {
+                consecutiveCount++;
+            }
+        }
+        bitMap.emplace_back(std::make_pair(lastValue, consecutiveCount));
+    }
+    return bitMap;
+}
+
+PGMBitMap decompressGrayscale(const CompressedPGMBitMap& compressedImg) {
+    PGMBitMap decompressed;
+
+    size_t pairCounter = 0;
+    HeightType currentRow = 0;
+    size_t idxWithinRow = 0;
+    auto currentEmptySlot = decompressed[currentRow].begin();
+
+    for (const auto& pair : compressedImg) {
+        pairCounter++;
+
+        if (pair.second > (width - idxWithinRow)) {
```

Comment:
Huge + for checking that :) I was also concerned about that, but at the end I have different problems and I skip this.